modules/Microsoft.App/managedEnvironments/deploy.bicep

```diff
@@ -0,0 +1,171 @@
+param logAnalyticsWorkspaceResourceId string
+
+@description('Optional. Location for all Resources.')
+param location string = resourceGroup().location
+
+@description('Optional. Tags of the resource.')
+param tags object = {}
+
+@description('Optional. Array of role assignment objects that contain the \'roleDefinitionIdOrName\' and \'principalId\' to define RBAC role assignments on this resource. In the roleDefinitionIdOrName attribute, you can provide either the display name of the role definition, or its fully qualified ID in the following format: \'/providers/Microsoft.Authorization/roleDefinitions/c2f4ef07-c644-48eb-af81-4b1b4947fb11\'.')
+param roleAssignments array = []
+
+@allowed([
+  'Consumption'
+  'Premium'
+])
+@description('Optional. Managed environment SKU.')
+param skuName string = 'Consumption'
+
+@description('Optional. Logs destination.')
+param logsDestination string = 'log-analytics'
+
+@description('Optional. Enable telemetry via a Globally Unique Identifier (GUID).')
+param enableDefaultTelemetry bool
+
+@description('Optional. Application Insights connection string used by Dapr to export Service to Service communication telemetry.')
+@secure()
+param daprAIConnectionString string = ''
+
+@description('Optional. Azure Monitor instrumentation key used by Dapr to export Service to Service communication telemetry.')
+@secure()
+param daprAIInstrumentationKey string = ''
+
+@description('Optional. CIDR notation IP range assigned to the Docker bridge, network. Must not overlap with any other provided IP ranges.')
+param dockerBridgeCidr string = ''
+
+@description('Optional. Resource ID of a subnet for infrastructure components. This subnet must be in the same VNET as the subnet defined in runtimeSubnetId. Must not overlap with any other provided IP ranges.')
+param infrastructureSubnetId string = ''
+
+@description('Optional. Boolean indicating the environment only has an internal load balancer. These environments do not have a public static IP resource. They must provide runtimeSubnetId and infrastructureSubnetId if enabling this property.')
+param internal bool = false
+
+@description('Optional. Configuration used to control the Environment Egress outbound traffic.')
+param vnetOutboundSettings object = {}
+
+@description('Optional. IP range in CIDR notation that can be reserved for environment infrastructure IP addresses. Must not overlap with any other provided IP ranges.')
+param platformReservedCidr string = ''
+
+@description('Optional. An IP address from the IP range defined by platformReservedCidr that will be reserved for the internal DNS server.')
+param platformReservedDnsIP string = ''
+
+@description('Optional. Resource ID of a subnet that Container App containers are injected into. This subnet must be in the same VNET as the subnet defined in infrastructureSubnetId. Must not overlap with any other provided IP ranges.')
+param runtimeSubnetId string = ''
```

Comment:
My original proposal was to either keep `infrastructureSubnetId` or `runtimeSubnetId` as they both have the same value. Plus this should be conditional if `internal` is `true`. Also suggested we have two test files, one for internal and one for external

Reply:
Any chance you could help me with that? As I don't know the service, I'm not sure how to properly name it, don't understand why those 2 have the same value (yet are two service properties), and how the test case would look like :o

Reply:
As discussed, we'll expose the `infrastructureSubnetId` as a parameter and then set the property `runtimeSubnetId` to be the value `infrastructureSubnetId` parameter.

Reply:
![image](https://user-images.githubusercontent.com/28486158/222373386-0407eb51-7b86-4aea-ab11-6f63352040ad.png)

Reply:
Removed